### PR TITLE
Cluster-autoscaler: add get nodes function to cloud provider interface

### PR DESCRIPTION
**Pull request by mwielgus**
This is needed to compare the list of nodes created by the cloud provider with the list of nodes registered in Kubernetes.

Ref: #2228 #2229

cc: @jszczepkowski @andrewsykim @piosz @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 1]
```diff
@@ -208,3 +198,32 @@ func (m *AwsManager) regenerateCache() error {
...
+	}
+	groups, err := m.service.DescribeAutoScalingGroups(params)
+	if err != nil {
+		glog.V(4).Infof("Failed ASG info request for %s: %v", name, err)
+		return nil, err
+	}
+	if len(groups.AutoScalingGroups) < 1 {
+		return nil, fmt.Errorf("Unable to get first autoscaling.Group for %s", name)
+	}
+	return groups.AutoScalingGroups[0], nil
```
Comment (andrewsykim) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: `return *groups.AutoScalingGroups[0], nil`?

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 2]
```diff
@@ -208,3 +198,32 @@ func (m *AwsManager) regenerateCache() error {
...
+}
+
+// GetMigNodes returns Asg nodes.
```
Comment (jszczepkowski) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: s/GetMigNodes/GetAsgNodes/

**Comment by jszczepkowski** [comment]
LGTM (after resolving comments)

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: Well, it compiles without *.

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: Done.

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go (hunk 1 above)
Comment (andrewsykim) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: lol ok

**Pull request merged**